src/OSPSuite.SimModel/include/SimModel/Species.h

```diff
@@ -19,12 +19,14 @@ class Species :
 {
 protected:
 	TObjectList<Formula> _rhsFormulaList;
+	//List of all formulas that have a reference to this species.
+	TObjectList<Formula> _allFormulaList;
```

Comment:
I would suggest to use just a "normal" std vector for `_allFormulaList`.
This will fix most (hopefuly: all) of the faling tests.
Loss of performance would be barely noticable

Reply:
Yes, this fixes the tests.